File: snippets/csharp/SendPayment.cs
using Breez.Sdk.Liquid;

public class SendPaymentSnippets
{
    public void GetCurrentLightningLimits(BindingLiquidSdk sdk)
    {
        // ANCHOR: get-current-pay-lightning-limits
        try
        {
            var currentLimits = sdk.FetchLightningLimits();
            Console.WriteLine($"Minimum amount, in sats: {currentLimits.send.minSat}");
            Console.WriteLine($"Maximum amount, in sats: {currentLimits.send.maxSat}");
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: get-current-pay-lightning-limits
    }

    public void PrepareSendPaymentLightningBolt11(BindingLiquidSdk sdk)
    {
        // ANCHOR: prepare-send-payment-lightning-bolt11
        // Set the bolt11 invoice you wish to pay
        var destination = "<bolt11 invoice>";
        try
        {
            var prepareResponse = sdk.PrepareSendPayment(new PrepareSendRequest(destination, null, null, null));

            // If the fees are acceptable, continue to create the Send Payment
            var sendFeesSat = prepareResponse.feesSat;
            Console.WriteLine($"Fees: {sendFeesSat} sats");
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: prepare-send-payment-lightning-bolt11
    }

    public void PrepareSendPaymentLightningBolt12(BindingLiquidSdk sdk)
    {
        // ANCHOR: prepare-send-payment-lightning-bolt12
        // Set the bolt12 offer you wish to pay
        var destination = "<bolt12 offer>";
        try
        {
            var optionalAmount = new PayAmount.Bitcoin(5000);
            var prepareRequest = new PrepareSendRequest(destination, optionalAmount, null, null);
            var prepareResponse = sdk.PrepareSendPayment(prepareRequest);
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: prepare-send-payment-lightning-bolt12
    }

    public void PrepareSendPaymentLiquid(BindingLiquidSdk sdk)
    {
        // ANCHOR: prepare-send-payment-liquid
        // Set the Liquid BIP21 or address you wish to pay
        var destination = "<Liquid BIP21 or address>";
        try
        {
            var optionalAmount = new PayAmount.Bitcoin(5000);
            var prepareResponse = sdk.PrepareSendPayment(new PrepareSendRequest(destination, optionalAmount, null, null));

            // If the fees are acceptable, continue to create the Send Payment
            var sendFeesSat = prepareResponse.feesSat;
            Console.WriteLine($"Fees: {sendFeesSat} sats");
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: prepare-send-payment-liquid
    }

    public void PrepareSendPaymentLiquidDrain(BindingLiquidSdk sdk)
    {
        // ANCHOR: prepare-send-payment-liquid-drain
        // Set the Liquid BIP21 or address you wish to pay
        var destination = "<Liquid BIP21 or address>";
        try
        {
            var optionalAmount = new PayAmount.Drain();
            var prepareResponse = sdk.PrepareSendPayment(new PrepareSendRequest(destination, optionalAmount, null, null));

            // If the fees are acceptable, continue to create the Send Payment
            var sendFeesSat = prepareResponse.feesSat;
            Console.WriteLine($"Fees: {sendFeesSat} sats");
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: prepare-send-payment-liquid-drain
    }

    public void SendPayment(BindingLiquidSdk sdk, PrepareSendResponse prepareResponse)
    {
        // ANCHOR: send-payment
        try
        {
            var optionalPayerNote = "<payer note>";
            var sendResponse = sdk.SendPayment(new SendPaymentRequest(prepareResponse, null, optionalPayerNote));
            var payment = sendResponse.payment;
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: send-payment
    }}

File: snippets/python/src/send_payment.py
import logging
from breez_sdk_liquid import BindingLiquidSdk, PayAmount, PrepareSendRequest, SendPaymentRequest, PrepareSendResponse

def fetch_pay_lightning_limits(sdk: BindingLiquidSdk):
    # ANCHOR: get-current-pay-lightning-limits
    try:
        current_limits = sdk.fetch_lightning_limits()
        logging.debug(f"Minimum amount, in sats {current_limits.send.min_sat}")
        logging.debug(f"Maximum amount, in sats {current_limits.send.max_sat}")
        return current_limits
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: get-current-pay-lightning-limits

def prepare_send_payment_lightning_bolt11(sdk: BindingLiquidSdk):
    # ANCHOR: prepare-send-payment-lightning-bolt11
    # Set the bolt11 invoice you wish to pay
    destination = "<bolt11 invoice>"
    try:
        prepare_response = sdk.prepare_send_payment(
            PrepareSendRequest(
                destination=destination
            )
        )

        # If the fees are acceptable, continue to create the Send Payment
        send_fees_sat = prepare_response.fees_sat
        logging.debug(f"Fees: {send_fees_sat} sats")
        return prepare_response
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: prepare-send-payment-lightning-bolt11

def prepare_send_payment_lightning_bolt12(sdk: BindingLiquidSdk):
    # ANCHOR: prepare-send-payment-lightning-bolt12
    # Set the bolt12 offer you wish to pay
    destination = "<bolt12 offer>"
    try:
        optional_amount = PayAmount.BITCOIN(5_000)

        prepare_response = sdk.prepare_send_payment(
            PrepareSendRequest(
                destination=destination,
                amount=optional_amount,
            )
        )

        return prepare_response
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: prepare-send-payment-lightning-bolt12

def prepare_send_payment_liquid(sdk: BindingLiquidSdk):
    # ANCHOR: prepare-send-payment-liquid
    # Set the Liquid BIP21 or Liquid address you wish to pay
    destination = "<Liquid BIP21 or address>"
    try:
        optional_amount = PayAmount.BITCOIN(5_000)
        prepare_response = sdk.prepare_send_payment(
            PrepareSendRequest(
                destination=destination,
                amount=optional_amount,
            )
        )

        # If the fees are acceptable, continue to create the Send Payment
        send_fees_sat = prepare_response.fees_sat
        logging.debug(f"Fees: {send_fees_sat} sats")
        return prepare_response
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: prepare-send-payment-liquid

def prepare_send_payment_liquid_drain(sdk: BindingLiquidSdk):
    # ANCHOR: prepare-send-payment-liquid-drain
    # Set the Liquid BIP21 or Liquid address you wish to pay
    destination = "<Liquid BIP21 or address>"
    try:
        optional_amount = PayAmount.DRAIN()
        prepare_response = sdk.prepare_send_payment(
            PrepareSendRequest(
                destination=destination,
                amount=optional_amount,
            )
        )

        # If the fees are acceptable, continue to create the Send Payment
        send_fees_sat = prepare_response.fees_sat
        logging.debug(f"Fees: {send_fees_sat} sats")
        return prepare_response
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: prepare-send-payment-liquid-drain

def send_payment(sdk: BindingLiquidSdk, prepare_response: PrepareSendResponse):
    # ANCHOR: send-payment
    optional_payer_note = "<payer note>"
    try:
        send_response = sdk.send_payment(
            SendPaymentRequest(
                prepare_response=prepare_response,
                payer_note=optional_payer_note
            )
        )
        payment = send_response.payment
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: send-payment
